sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ParDo.java

```diff
@@ -683,6 +686,18 @@ public static DoFnSchemaInformation getDoFnSchemaInformation(
                 (SchemaCoder<?>) input.getCoder(), accessDescriptor, selectedSchema, elementT);
       }
     }
+    if (input.hasSchema()) {
```

Comment:
unnecessary - we wouldn't hit this codepath if !input.hasSchema

Reply:
Removed.